### PR TITLE
Enable active scalar selection for the ModuleSlice

### PR DESCRIPTION
**Pull request by alesgenova**
This PR brings the slice to feature parity with the ModuleVolume.
For a multi-scalar datasource, the active scalars shown by the slice can either
be in synch with what is selected upstream on the datasource, or they
can be locked to a specific array. The user can freely go back and
forth.

At the implementation level, I created a subclass of the
vtkTrivialProducer, the vtkActiveScalarsProducer.
vtkActiveScalarsProducer stores internally a shallow copy of the
vtkImageData that is set as output. The active scalars can be set on the
shallow copy with no repercussion on the original data.

Signed-off-by: Alessandro Genova <alessandro.genova@kitware.com>

Please provide an overview of what this pull request does, and reference any
relevant issues that are addressed. Once submitted you are encouraged to seek
review of the code, and to check the continuous integration results.

By submitting this pull request I agree to the terms of the
[Developer Certificate or Origin][dco].

  [dco]: https://developercertificate.org/

### CODE REVIEW

File: tomviz/CMakeLists.txt [hunk 1]
```diff
@@ -209,6 +209,7 @@ set(SOURCES
   vtkCustomPiecewiseControlPointsItem.h
   vtkLengthScaleRepresentation.h
   vtkLengthScaleRepresentation.cxx
+  vtkActiveScalarsProducer.cxx
```
Comment (cryos) [inline tomviz/CMakeLists.txt]: Add the header here too, as with the other classes.

File: tomviz/vtkActiveScalarsProducer.cxx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+#include <vtkGarbageCollector.h>
+#include <vtkImageData.h>
+#include <vtkObjectFactory.h>
+#include <vtkPointData.h>
+
+vtkStandardNewMacro(vtkActiveScalarsProducer);
+
+vtkActiveScalarsProducer::vtkActiveScalarsProducer()
+{
+  this->OriginalOutput = nullptr;
```
Comment (cryos) [inline tomviz/vtkActiveScalarsProducer.cxx]: We should use C++11 and assign nullptr directly at declaration in the header for both of these.

File: tomviz/vtkActiveScalarsProducer.cxx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+vtkActiveScalarsProducer::~vtkActiveScalarsProducer()
+{
+  this->SetOutput(nullptr);
+}
+
+//----------------------------------------------------------------------------
+void vtkActiveScalarsProducer::SetOutput(vtkDataObject* newOutput)
+{
+  vtkDataObject* oldOutput = this->OriginalOutput;
```
Comment (cryos) [inline tomviz/vtkActiveScalarsProducer.cxx]: auto

File: tomviz/vtkActiveScalarsProducer.cxx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    if (oldOutput) {
+      oldOutput->UnRegister(this);
+    }
+    this->Modified();
+  }
+}
+
+void vtkActiveScalarsProducer::SetActiveScalars(char* name)
+{
+  vtkImageData* data = vtkImageData::SafeDownCast(this->Output);
```
Comment (cryos) [inline tomviz/vtkActiveScalarsProducer.cxx]: auto

File: tomviz/vtkActiveScalarsProducer.cxx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  if (data) {
+    data->GetPointData()->SetActiveScalars(name);
+    data->Modified();
+  }
+}
+
+vtkMTimeType vtkActiveScalarsProducer::GetMTime()
+{
+  vtkMTimeType mtime = this->Superclass::GetMTime();
```
Comment (cryos) [inline tomviz/vtkActiveScalarsProducer.cxx]: auto

File: tomviz/vtkActiveScalarsProducer.cxx [hunk 6]
```diff
@@ -0,0 +1,78 @@
+#include "vtkActiveScalarsProducer.h"
```
Comment (cryos) [inline tomviz/vtkActiveScalarsProducer.cxx]: Missing copyright lines.

File: tomviz/vtkActiveScalarsProducer.h [hunk 7]
```diff
@@ -0,0 +1,33 @@
+#ifndef vtkActiveScalarsProducer_h
```
Comment (cryos) [inline tomviz/vtkActiveScalarsProducer.h]: Missing copyright lines.

**Comment by alesgenova** [comment]
@cryos I have addressed the issues you mentioned

**Pull request merged**